x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/authz/store/ReservedRolesStore.java

```diff
@@ -124,11 +124,19 @@
                                         .indices(".monitoring-*").privileges("read", "read_cross_cluster").build(),
                                 RoleDescriptor.IndicesPrivileges.builder()
                                         .indices(".management-beats").privileges("create_index", "read", "write").build(),
-                                // .apm-* is for APM's agent configuration and custom link index creation
+                                // APM agent configuration
                                 RoleDescriptor.IndicesPrivileges.builder()
                                         .indices(".apm-agent-configuration").privileges("all").build(),
+                                // APM custom link index creation
                                 RoleDescriptor.IndicesPrivileges.builder()
                                         .indices(".apm-custom-link").privileges("all").build(),
+                                // APM telemetry queries APM & ML anomalies indices in kibana task runner
+                                RoleDescriptor.IndicesPrivileges.builder()
+                                    .indices("apm-*")
```

Comment:
Most of the indices referenced in this file is prefixed with a dot. So just wanna make sure this one is indeed without it.

Reply:
yes it's intentionally referring to the apm data indices

Reply:
@ogupte In a previous discussion, of which you have been part of, see https://github.com/elastic/elasticsearch/pull/50051#issuecomment-564656084, Brandon laid out the design of security privileges.

According to it, system roles such as `kibana_system`, which are used by Kibana system tasks/jobs for internal house keeping, must not be granted privileges against end user data. The administrator must at all times be in control of which users and services have access to the end user's data.

If a system task needs to deal with user data, it must get consent from the user, and afterwards obtain API keys on the user's behalf (see https://github.com/elastic/elasticsearch/pull/52886). The end users should be able to revoke the system user's access to his data at any time.

IIUC this PR grants privileges to user data to the `kibana_system` role, going against the restrictions described above.